src/decoder/mod.rs

```diff
@@ -429,6 +458,36 @@ impl<R: Read + Seek> Decoder<R> {
             0 => None,
             n => Some(n),
         };
+        // A special BigTiff format is generated by ScanImage and is identified
+        // and (partially) parsed under this if statement. It's identified by
+        // matching on a "magic number" that ScanImage defines [here](http://scanimage.vidriotechnologies.com/display/SI2019/ScanImage+BigTiff+Specification#ScanImageBigTiffSpecification-ScanImageStaticMetadataSection).
+        if self.bigtiff {
+            let mut si_magic_number_buffer = [0u8; 4];
+            self.peek_n_bytes(&mut si_magic_number_buffer)?;
+            let magic_num = u32::from_le_bytes(si_magic_number_buffer);
+            if magic_num == 117637889 {
+                &mut self.parse_si_header_metadata()?;
+            }
+        }
+        Ok(())
+    }
+
+    fn parse_si_header_metadata(&mut self) -> TiffResult<()> {
+        // Read the magic number again and discard it
+        self.read_long()?;
+        let tif_version = self.read_long()?;
+        let nonvar_frame_data_length = self.read_long()?;
+        let roi_group_data_length = self.read_long()?;
+        let mut nonvar_frame_data = vec![0u8; usize::try_from(nonvar_frame_data_length)?];
+        self.reader.read_exact(&mut nonvar_frame_data)?;
+        let mut roi_group_data = vec![0u8; usize::try_from(roi_group_data_length)?];
+        self.reader.read_exact(&mut roi_group_data)?;
+        // Parse the read bytes into UTF8 and into the metadat object
+        let nonvar_frame_data = String::from_utf8(nonvar_frame_data)?;
+        let roi_group_data = String::from_utf8(roi_group_data)?;
```

Comment:
If you want these to error with `InvalidScanImageMetadataEncoding` then you can use:

```suggestion
        let nonvar_frame_data = String::from_utf8(nonvar_frame_data)
            .map_err(|_| InvalidScanImageMetadataEncoding)?;
        let roi_group_data = String::from_utf8(roi_group_data)
            .map_err(|_| InvalidScanImageMetadataEncoding)?;
```

Reply:
You could add a field to the variant to keep the underlying conversion error—and display it if you want.